[stable/sentry] resource limitations and file upload fix

ahmadalli opened the pull request:
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

ahmadalli committed:
change resource limitations to proper values
sentry will crash otherwise

Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

ahmadalli committed:
change resource limitations to proper values
sentry will crash otherwise

Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

ahmadalli committed:
change service type
ClusterIP is more common on charts than load balancer

Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

ahmadalli committed:
change service type
ClusterIP is more common on charts than load balancer

Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

ahmadalli committed:
file upload fix

Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

ahmadalli committed:
file upload fix

Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

ahmadalli committed:
version bump

Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

ahmadalli committed:
version bump

Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

ahmadalli committed:
share sentry files between web and workers

Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

ahmadalli committed:
share sentry files between web and workers

Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

okgolove commented on stable/sentry/values.yaml:
```diff
@@ -14,7 +14,7 @@ web:
   resources:
     limits:
-      cpu: 500m
+      cpu: 4000m
```
It is too big limit, isn't it?

ahmadalli replied:
as sentry [stated](https://docs.sentry.io/server/installation/), sentry is not designed to run on low resource systems. we faced too many container crashes on our productions systems with the previous limit which was resolved by increasing the resource limits

okgolove replied:
I'm sorry you faced the crashes.
But your requirements pushes user to have at least 2 x c5.2xlarge (for worker and web).
To be honest I'd like not to specify default resources.

ahmadalli replied:
you mean it's ok to remove them completely? or just keep the old limitations?
also does setting `limits` (and not `requests`) makes user to have the amount of resources that are stated on `limites` section?

okgolove replied:
It doesn't, but:
```When Pods use more memory than their limit, a process that is using the most amount of memory, inside one of the pod's containers, will be killed by the kernel```
https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node/resource-qos.md#qos-classes

I think we should save old values, I'll rework it later.

ahmadalli replied:
the point is sentry seems useless with that low amount of resources. why should we have them as default and ship a broken chart to the users?

okgolove replied:
I wouldn't say it's broken.
For example, I'm using it with default resources and it works properly.
https://imgur.com/a/hy9hY2C
https://imgur.com/a/AMFtHFW

But if you're totally sure we shouldn't use the limits were specified before then please remove them at all and set:
```
resources: {}
  # We usually recommend not to specify default resources and to leave this as a conscious
  # choice for the user. This also increases chances charts run on environments with little
  # resources, such as Minikube. If you do want to specify resources, uncomment the following
  # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
  # limits:
  #  cpu: 100m
  #  memory: 128Mi
  # requests:
  #  cpu: 100m
  #  memory: 128Mi
```

ahmadalli replied:
I've changed it to empty resources

okgolove commented on stable/sentry/values.yaml:
```diff
@@ -58,7 +58,7 @@ worker:
     limits:
       cpu: 300m
-      memory: 500Mi
+      memory: 8196Mi
```
The same idea, why 8Gb?

ahmadalli replied:
the same as above :)

okgolove commented on stable/sentry/templates/workers-deployment.yaml:
```diff
@@ -113,9 +113,18 @@ spec:
         - mountPath: /etc/sentry
           name: config
           readOnly: true
+        - mountPath: {{ .Values.persistence.filestore_dir }}
```
Let's follow CamelCase practice.

ahmadalli replied:
I didn't change the name. (it was added on initial sentry commit https://github.com/helm/charts/commit/cb0060835763bcdd68fb080ab4799f36a1033fa1) I just added the same volume to the worker deployments

maorfr replied:
Always leave the code cleaner then how you found it? :)

ahmadalli replied:
I'm on it :)

ahmadalli replied:
also the value need to be passed as `SENTRY_FILESTORE_DIR` env. I'm adding it too

ahmadalli replied:
is it ok to remove this value and hardcode it? I don't see any case that changing `SENTRY_FILESTORE_DIR` from `/var/lib/sentry/files` (which is inside the container) to other values does any help

okgolove replied:
I think in this case it's ok.

okgolove commented on stable/sentry/Chart.yaml:
```diff
@@ -1,7 +1,7 @@
 description: Sentry is a cross-platform crash reporting and aggregation platform.
 name: sentry
-version: 1.3.5
+version: 1.3.6
```
1.4.0

ahmadalli replied:
I've bumped the version

ahmadalli committed:
remove worker resource limits

Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

ahmadalli committed:
remove worker resource limits

Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

ahmadalli committed:
hardcore filestore directory variable

Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

ahmadalli committed:
hardcore filestore directory variable

Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

ahmadalli committed:
chart image update

Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

ahmadalli committed:
chart image update

Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

Commit:
[stable/prometheus-operator] allow definition of affinity for alertmanagerSpec/prometheusSpec (#12936)

* [stable/prometheus-operator] add affinity configuration

Extends the prometheus-operator to allow the definition of additional
affinity rules for alertmanager and prometheus without breaking the
existing podAntiAffinity functionality.

Signed-off-by: Lukas Grossar <lukas.grossar@adfinis-sygroup.ch>

* [stable/prometheus-operator] update ci/test-values.yaml

Got forgotten in some other PRs.

Signed-off-by: Lukas Grossar <lukas.grossar@adfinis-sygroup.ch>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

davidkarlsen committed:
[stable/external-dns] Upgrade to latest release (#14004)

Signed-off-by: David J. M. Karlsen <david@davidkarlsen.com>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

alemorcuq committed:
[stable/magento] Add Elasticsearch component (#14023)

Signed-off-by: Alejandro Moreno <amoreno@bitnami.com>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

batazor committed:
[stable/node-red] add member (#13928)

Signed-off-by: Login Victor <batazor111@gmail.com>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

komljen committed:
SC-4770 Remove region info from Helm chart (#14024)

Signed-off-by: Alen Komljen <alen.komljen@live.com>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

linki committed:
[stable/chaoskube] Update to v0.14.0 (#14027)

Signed-off-by: Martin Linkhorst <martin.linkhorst+github.com@posteo.de>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

anasinnyk committed:
[stable/atlantis] added possibility to override command for container (#13963)

* feat(atlantis): added possibillity to override command for container

Signed-off-by: Andrii Nasinnyk <anasinnyk@macpaw.com>

* feat(atlantis): added possibillity to override command for container

Signed-off-by: Andrii Nasinnyk <anasinnyk@macpaw.com>

* feat(atlantis): added possibillity to override command for container

Signed-off-by: Andrii Nasinnyk <anasinnyk@macpaw.com>

* docs: update command description

Co-Authored-By: Luke Kysow <1034429+lkysow@users.noreply.github.com>
Signed-off-by: Andrii Nasinnyk <anasinnyk@macpaw.com>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

amine7536 committed:
[stable/prometheus-operator]: add EnableAdminAPI option (#14030)

Signed-off-by: Amine Benseddik <amine@qonto.eu>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

verwilst committed:
[stable/nginx-ingress] Upgrade defaultbackend to 1.5 (#14026)

* upgrade defaultbackend

Signed-off-by: Bart Verwilst <bart@verwilst.be>

* update readme

Signed-off-by: Bart Verwilst <bart@verwilst.be>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

Commit:
[stable/prometheus-adapter] Add myself to OWNERS (#13887)

Signed-off-by: Steven Sheehy <ssheehy@firescope.com>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

lostick committed:
[stable/prometheus-operator] Bump alertmanager to 0.17.0 (#14035)

Signed-off-by: Laurent Godet <mevoyasf@gmail.com>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

alemorcuq committed:
[stable/magento] Disable Elasticsearch by default (#14042)

Signed-off-by: Alejandro Moreno <amoreno@bitnami.com>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

gordcorp committed:
[stable/sentry] Add post-upgrade hook to db-init job (#13845)

* Add post-upgrade hook to db-init job

Signed-off-by: Brendan Gordon <brendan.gordon@digital.gov.au>

* Incrementing  the versiono

Signed-off-by: Matt Farina <matt@mattfarina.com>

* Bump version

Signed-off-by: Brendan Gordon <brendan.gordon@digital.gov.au>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

eh-am committed:
add namespace metadata (#13956)

this is needed for when helm is used purely as a templating tool,
since helm template does not add the namespace
helm/helm#3553

Signed-off-by: eduardo aleixo <eduardoaleixomartins@gmail.com>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

Commit:
stable/odoo: update to 12.0.20190515 (#14025)

Signed-off-by: Bitnami Containers <containers@bitnami.com>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

lostick committed:
[stable/prometheus] Bump node-exporter to 0.18.0 (#14036)

Signed-off-by: Laurent Godet <mevoyasf@gmail.com>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

Commit:
stable/cockroachdb Updated README.md w/ upgrade instructions (#14029)

Updated README to direct to CockroachDB documentation on the Helm chart, and to provide updated upgrade guidance.

Checklist

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

[x] DCO signed
[x] Variables are documented in the README.md
[x] Title of the PR starts with chart name (e.g. [stable/chart])

Signed-off-by: Keith McClellan <keith.mcclellan@gmail.com>

	modified:   Chart.yaml

Signed-off-by: Keith McClellan <keith.mcclellan@gmail.com>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

naseemkullah committed:
[stable/elasticsearch] Run as non-root (#13429)

* Various

Signed-off-by: Naseem <naseemkullah@gmail.com>

* Option to run as elasticsearch, not root

Signed-off-by: Naseem <naseemkullah@gmail.com>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

Commit:
velero: add me as maintainer (#14031)

Signed-off-by: Hector Fernandez <hfernandez@mesosphere.com>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

batazor committed:
[stable/kube-state-metrics] Solve #13961 (#13995)

Signed-off-by: Login Victor <batazor111@gmail.com>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

alemorcuq committed:
[stable/magento] Add values-production.yaml (#14051)

* [stable/magento] Add values-production.yaml

Signed-off-by: Alejandro Moreno <amoreno@bitnami.com>

* Fix wording

Signed-off-by: Alejandro Moreno <amoreno@bitnami.com>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

cameronattard committed:
[stable/cerebro] add ingress labels (#14053)

Signed-off-by: Cameron Attard <cameron.attard@siteminder.com>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

Commit:
[stable/weave-scope] update weave-scope to 1.11.0 (#13949)

* [stable/weave-scope] update weave-scope to 1.10.2

Signed-off-by: Daniel Holbach <daniel@weave.works>

* [stable/weave-scope] update chart version as well

Signed-off-by: Daniel Holbach <daniel@weave.works>

* update to 1.11.0

Signed-off-by: Daniel Holbach <daniel@weave.works>

* add OWNERS file for weave-scope

	Bryan Boreham and Filip Barl are maintainers of Weave Scope
	(https://github.com/weaveworks/scope/blob/master/MAINTAINERS)

	Daniel Holbach helps out with releases and community management
	occasionally. We all work for Weaveworks.

	Thanks Maor Friedman for this suggestion!
	(https://kubernetes.slack.com/archives/C6E3XH1ED/p1558452865042400?thread_ts=1558452289.041500&cid=C6E3XH1ED)

Signed-off-by: Daniel Holbach <daniel@weave.works>

* add Daniel, Filip and Bryan as maintainers in Charts.yaml

Signed-off-by: Daniel Holbach <daniel@weave.works>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

Yannig committed:
Handle priorityClassName for fluentd-cloudwatch. (#13847)

Signed-off-by: Yannig Perré <yannig.perre@aios.sh>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

foot committed:
[stable/weave-cloud] Upgrades bundled kubectl to 1.11.10 (#13994)

Signed-off-by: Simon Howe <simon@weave.works>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

verwilst committed:
[stable/dex] run jobs as nobody instead of root (#14057)

* run as nobody instead of root

Signed-off-by: Bart Verwilst <bart@verwilst.be>

* increment version

Signed-off-by: Bart Verwilst <bart@verwilst.be>

* make securityContext configurable

Signed-off-by: Bart Verwilst <bart@verwilst.be>

* this change warrants a minor release bump imo

Signed-off-by: Bart Verwilst <bart@verwilst.be>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

Commit:
stable/ghost: update to 2.22.2 (#14060)

Signed-off-by: Bitnami Containers <containers@bitnami.com>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

maorfr committed:
[incubator/kube-registry-proxy] add apiVersion (#13895)

* [incubator/kube-registry-proxy] add apiVersion

Signed-off-by: Maor Friedman <maor.friedman@redhat.com>

* [incubator/kube-registry-proxy] add maintainer

Signed-off-by: Maor Friedman <maor.friedman@redhat.com>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

Commit:
[stable/rocketchat] Add a warning about upgrading to 1.x (#14068)

Signed-off-by: Simon Rüegg <simon.ruegg@vshn.ch>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

AndiDog committed:
[stable/nginx-ingress] Evaluate tcp/udp as template (#14063)

Signed-off-by: Andreas Sommer <andreas.sommer87@googlemail.com>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

smcavallo committed:
[incubator/schema-registry] allow custom service labels (#14071)

* allow custom service labels for schema registry
Signed-off-by: S.Cavallo <smcavallo@hotmail.com>

* bump chart version
Signed-off-by: S.Cavallo <smcavallo@hotmail.com>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

mmiller1 committed:
allow adding targetLabels to servicemonitors (#14072)

Signed-off-by: Matt Miller <mamiller@rosettastone.com>

update-ci

Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

anothertobi committed:
[stable/prometheus-operator] Incorporate upstream changes in clusterrole.yaml (#13759)

* Allow the prometheus-operator to delete services

This change was introduced in the prometheus-operator repository in
https://github.com/coreos/prometheus-operator/pull/2319.

Signed-off-by: Tobias Nehrlich <tobias.nehrlich@vshn.ch>

* Add myself as a maintainer

Since we are using this operator @vshn I will keep maintaining the
stable chart.

Signed-off-by: Tobias Nehrlich <tobias.nehrlich@vshn.ch>

* Bump chart version

Signed-off-by: Tobias Nehrlich <tobias.nehrlich@vshn.ch>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

cw-sakamoto committed:
[incubator/aws-alb-ingress-controller] Add volumes/volumeMounts option to aws-alb-ingress-controller (#13641)

* add volume option to aws-alb-ingress-controller

Signed-off-by: cw-sakamoto <sakamoto@chatwork.com>

* modified comment spaces in values.yaml

Signed-off-by: cw-sakamoto <sakamoto@chatwork.com>
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

ahmadalli committed:
Merge branch 'sentry-patch' of github.com:ahmadalli/charts into sentry-patch

Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>
